src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -214,42 +215,63 @@ private Callback kafkaSendCallback(final BatchItem item, final CountDownLatch do
 
     @Override
     public List<TopicPartition> listPartitions(final String topicId) throws ServiceUnavailableException {
-
         try (final Consumer<String, String> consumer = kafkaFactory.getConsumer()) {
-
             final List<org.apache.kafka.common.TopicPartition> kafkaTPs = consumer
                     .partitionsFor(topicId)
                     .stream()
                     .map(p -> new org.apache.kafka.common.TopicPartition(topicId, p.partition()))
                     .collect(toList());
 
-            consumer.assign(kafkaTPs);
+            return toNakadiTopicPartition(consumer, kafkaTPs);
+        } catch (final Exception e) {
+            throw new ServiceUnavailableException("Error occurred when fetching partitions offsets", e);
+        }
+    }
 
-            final org.apache.kafka.common.TopicPartition[] tpArray =
-                    kafkaTPs.toArray(new org.apache.kafka.common.TopicPartition[kafkaTPs.size()]);
+    @Override
+    public List<TopicPartition> listPartitions(final Set<String> topics) throws ServiceUnavailableException {
+        try (final Consumer<String, String> consumer = kafkaFactory.getConsumer()) {
+            final List<org.apache.kafka.common.TopicPartition> kafkaTPs = consumer.listTopics().entrySet().stream()
+                    .filter(entry -> topics.contains(entry.getKey()))
+                    .flatMap(entry -> entry.getValue().stream()
+                            .map(partitionInfo -> new org.apache.kafka.common.TopicPartition(entry.getKey(),
+                                    partitionInfo.partition()))
+                    )
+                    .collect(toList());
 
-            consumer.seekToBeginning(tpArray);
-            final Map<Integer, Long> earliestOffsets = getPositions(consumer, kafkaTPs);
+            return toNakadiTopicPartition(consumer, kafkaTPs);
+        } catch (final Exception e) {
+            throw new ServiceUnavailableException("Error occurred when fetching partitions offsets", e);
+        }
+    }
 
-            consumer.seekToEnd(tpArray);
-            final Map<Integer, Long> latestOffsets = getPositions(consumer, kafkaTPs);
+    private List<TopicPartition> toNakadiTopicPartition(final Consumer<String, String> consumer,
```

Comment:
I think the name of method doesn't fully reflect what happens.
I think the better name can be something like `getPartitionsBounds` or something like that